### DISCUSSION TITLE
gsudo with "/etc/sudoers" functionality?

### DISCUSSION THREAD

**Discussion by igrokit** [gerardog/gsudo#205] [Ideas]
In Linux, I can give any user the possibility to execute root commands by adding them to /etc/sudoers. This is of course somewhat simplified. The "sudoers" file in Linux has quite some possibilities for restricting or allowing things the user can do..
The user then has to enter his own password to gain admin privileges. 

Would something similar be possible for gsudo?
Of course there would have to be some restrictions to be applied as well. 
I.e. That user should not be able to change the gsudo configuration and other restrictions that make sense. 

Or perhaps there is another nice utility that allows a normal user to gain admin privileges for a short time?
Dan

**Comment by gerardog** [comment]
Hi igrokit,

The Microsoft way to do this would be to create pre configured, Scheduled Task.

On the other hand, I would love to add the sudoers functionality and has been in my mind since the very first version. (It's mentioned in the readme from already [2 years](https://github.com/gerardog/gsudo/commit/af68b2fc33c80bcc018edecbdd26ad6c21d50f06#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R199) ago). It's not a small task: It would require gsudo running as a proper windows service. (Opposite of the current promise of being a portable app that auto elevates on demand...) It means a different design, securing the service and, as you said, secured configuration.

This is not small effort, and I lack the time... 

... and what about the motivation?... So far, gsudo has been all voluntary work. I'm grateful of all the donations received, but they just helped cover past expenses. I've had thoughts of doing a 'pro' paid version to see if that could become profitable. And not sure maybe this could be it since: 1) it needs to be a different (non-portable) version, 2) It provides additional value to the user, 3) requires major effort to develop,  

Feature Ideas I've collected that it should have are:
 * sudoers (pre-authorize specific commands)
 * to craft custom tokens with local admin privileges for regular users ( likey to granting admin priv to a regular user temporarily and for one process (tree) only.)
 * custom challenges, like passphrase, windows hello (face or fingerprint)
 * maybe even replacing windows UAC dialog with a new custom window.

I'm being totally honest here. For example, I have no idea if there is demand for such tool.
And I sincerely hope you don't find disappointing that I am looking for ways to get retribution for my work! 

We can discuss possible use cases if you want. Like, how would you expect it to be ?

**Reply by SZRabinowitz** [reply]
Hi @gerardog , just thought I'd give my two cents. 

First of all thanks for all your hard work and your great tool. I loved it since day 1.

If you made a paid version, I wouldn't pay for it because:
1. I cant really afford it (I am a student, I don't have a credit card)
2. I don't really need those features.

But I think it is a great idea to try and make money, although I would be sad if you removed existing features of the free version. But, nobody owns you, and you have the right to do whatever you want. I am not entitled to anything, I just enjoy a free tool :). 

I have you on the list of projects to donate to if I get money (dont get your hopes up, I probably will not).

**Reply by gerardog** [reply]
Thank you for you honesty. Rest assured: If I ever create a paid version, I wont remove any existing features of the free version.
